tests/D2L.CodeStyle.Analyzers.Test/SpecTestsProvider.cs

```diff
@@ -0,0 +1,136 @@
+					builder.Add( field.Name, diagnosticDescriptor );
+				}
+			}
+
+			return builder.ToImmutable();
+		}
+
+		private static ImmutableArray<MetadataReference> GetMetadataReferences() {
+
+			var builder = ImmutableArray.CreateBuilder<MetadataReference>();
+
+			builder.AddAssemblyOf( typeof( object ) );                  // mscorlib
+			builder.AddAssemblyOf( typeof( Uri ) );                     // System
+			builder.AddAssemblyOf( typeof( Enumerable ) );              // System.Core
+			builder.AddAssemblyOf( typeof( Annotations.Because ) );     // D2L.CodeStyle.Annotations
+
+			Assembly systemRuntime = Assembly.Load( "System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" );
+			builder.Add( MetadataReference.CreateFromFile( systemRuntime.Location ) );
+
+			return builder.ToImmutable();
+		}
+
+		private static void AddAssemblyOf( this ImmutableArray<MetadataReference>.Builder builder, Type type ) {
+			MetadataReference reference = MetadataReference.CreateFromFile( type.Assembly.Location );
+			builder.Add( reference );
+		}
+
+		private static ImmutableArray<AdditionalText> GetAdditionalFiles() {
+
+			var builder = ImmutableArray.CreateBuilder<AdditionalText>();
+
+			Assembly testAssembly = Assembly.GetExecutingAssembly();
+			foreach( string resourcePath in testAssembly.GetManifestResourceNames() ) {
+
+				if( !resourcePath.EndsWith( "AllowedList.txt" ) ) {
+					continue;
+				}
+
+				string virtualPath = Regex.Replace(
+					resourcePath,
+					@"^.*\.(?<allowedListName>[^\.]*)\.txt$",
+					@"${allowedListName}.txt"
+				);
+
+				string text = ReadEmbeddedResourceAsString( testAssembly, resourcePath );
+
+				builder.Add( new AdditionalFile( virtualPath, text ) );
+			}
+
+			return builder.ToImmutable();
+		}
+
+		private static IEnumerable<(string SpecName, string Source)> GetEmbeddedSpecTests() {
+
+			Assembly assembly = Assembly.GetExecutingAssembly();
+
+			foreach( string specFilePath in assembly.GetManifestResourceNames() ) {
+
+				if( !specFilePath.EndsWith( ".cs" ) ) {
+					continue;
+				}
+
+				// The file foo/bar.baz.cs has specName bar.baz
+				string specName = Regex.Replace(
+					specFilePath,
+					@"^.*\.(?<specName>[^\.]*)\.cs$",
+					@"${specName}"
+				);
+
+				string source = ReadEmbeddedResourceAsString( assembly, specFilePath );
+
+				yield return (specName, source);
+			}
+		}
+
+		private sealed class AdditionalFile : AdditionalText {
+
+			private readonly string m_text;
+
+			public AdditionalFile(
+				string path,
+				string text
+			) {
+				Path = path;
+				m_text = text;
+			}
+
+			public override string Path { get; }
+			public override SourceText GetText( CancellationToken cancellationToken = default ) => SourceText.From( m_text, Encoding.UTF8 );
+		}
+
+		private static string ReadEmbeddedResourceAsString( Assembly assembly, string name ) {
+
+			using( Stream stream = assembly.GetManifestResourceStream( name ) )

```

Comment:
nit: there are statements now (`using var foo = ...`)

Reply:
Good catch.  It was from here:

![image](https://user-images.githubusercontent.com/12101120/153058364-96307f66-94dd-4d5c-9310-16053ebc0f38.png)

Reply:
I'll fix in a follow up PR